src/components/VideoControl.tsx

```diff
@@ -0,0 +1,110 @@
+import { FC, useEffect, useState, useRef, MutableRefObject } from 'react';
+
+import * as handPose from '@tensorflow-models/handpose';
+import '@tensorflow/tfjs-backend-webgl';
+
+import { Direction } from '../types';
+
+type VideoControlProps = {
+  onDirectionChange: (direction: Direction) => void;
+  isMovingRef: MutableRefObject<boolean>;
+  WIDTH: number;
+  HEIGHT: number;
+  isVideoEnabled: boolean;
+};
+
+export const VideoControl: FC<VideoControlProps> = ({
+  onDirectionChange,
+  isMovingRef,
```

Comment:
If `VideControl` should be a general-purpose control it should not contain `isMovingRef` that is specific to this game.

Considering above `useHandleButtons` should not use `isMovingRef` too.

I recommend moving `if (isMovingRef.current) return;` check to the beginning of `udpateState` function that will solve the issue.